src/segmantic/seg/monai_unet.py

```diff
@@ -803,3 +809,149 @@ def cross_validate(
+        #        output_transform=from_engine(["pred", "label"]),
+        #    )
+        # },
+    )
+    evaluator.run()
+
+
+def ensemble_creator(
+    model_files: List[Path],
+    test_images: List[Path],
+    test_labels: Optional[List[Path]] = None,
+    output_dir: Path = None,
+    tissue_dict: Dict[str, int] = None,
+    spacing: Sequence[float] = [],
+    combination_mode: str = "SelectBest",
+    gpu_ids: List[int] = [],
+):
+    device = make_device(gpu_ids)
+
+    models = [Net.load_from_checkpoint(str(ckpt_path)) for ckpt_path in model_files]
+
+    print(tissue_dict)
+
+    if test_labels:
+        assert len(test_images) == len(test_labels)
+        test_files = [
+            {"image": i, "label": l} for i, l in zip(test_images, test_labels)
+        ]
+    else:
+        test_files = [{"image": i} for i in test_images]
+
+    pre_transforms = Compose(
+        [
+            models[0].default_preprocessing(
+                keys=["image", "label"] if test_labels else ["image"],
+                spacing=spacing,
+            ),
+            # AsDiscreted(keys="label", to_onehot=models[0].num_classes),
+        ]
+    )
+
+    # data loader
+    test_loader = DataLoader(
+        Dataset(
+            data=test_files,
+            transform=pre_transforms,
+        ),
+        batch_size=1,
+        num_workers=0,
+    )
+
+    # save predicted labels
+    save_transforms: List[MapTransform] = []
+    if output_dir:
+        os.makedirs(output_dir, exist_ok=True)
+
+        save_transforms = [
+            SaveImaged(
+                keys="pred",
+                output_dir=output_dir,
+                output_postfix="seg",
+                resample=False,
+                separate_folder=False,
+                print_log=False,
+                writer="ITKWriter",
+            )
+        ]
+    if combination_mode == "Mean":
+        mean_post_transforms = Compose(
+            [
+                EnsureTyped(keys=["pred0", "pred1", "pred2"]),
+                MeanEnsembled(
+                    keys=["pred0", "pred1", "pred2"],
+                    output_key="pred",
+                    # in this particular example, we use validation metrics as weights
+                    weights=[0.95, 0.94, 0.95],
+                ),
+                # Activationsd(keys="pred", sigmoid=True),
+                AsDiscreted(keys="pred", threshold=0.5),
+            ]
+            + save_transforms
+        )
+        ensemble_evaluate(
+            mean_post_transforms, models, device=device, test_loader=test_loader
+        )
+    elif combination_mode == "Vote":
+        vote_post_transforms = Compose(
+            [
+                EnsureTyped(keys=["pred0", "pred1", "pred2"]),
+                # Activationsd(
+                #    keys=["pred0", "pred1", "pred2"], sigmoid=True
+                # ),
+                # transform data into discrete before voting
+                AsDiscreted(keys=["pred0", "pred1", "pred2"], threshold=0.5),
```

Comment:
`threshold=0.5` probably would require to use `Activationsd` above. But my (intuition) preference would be for `argmax=True` instead of thresholding

Reply:
makes sense. the mean and vote functions i still need to understand a bit better as well because the results were a bit strange.